refactor(tui2): make transcript line metadata explicit

joshka-oai opened the pull request:
This is a pure refactor only change.

Replace the flattened transcript line metadata from `Option<(usize, usize)>` to an explicit
`TranscriptLineMeta::{CellLine { cell_index, line_in_cell }, Spacer}` enum.

This makes spacer rows unambiguous, removes “tuple semantics” from call sites, and keeps the
scroll anchoring model clearer and aligned with the viewport/history design notes.

Changes:
- Introduce `TranscriptLineMeta` and update `TranscriptScroll` helpers to consume it.
- Update `App::build_transcript_lines` and downstream consumers (scrolling, row classification, ANSI rendering).
- Refresh scrolling module docs to describe anchors + spacer semantics in context.
- Add tests and docs about the behavior

Tests:
- just fmt
- cargo test -p codex-tui2 tui::scrolling

Manual testing:
- Scroll the inline transcript with mouse wheel + PgUp/PgDn/Home/End, then resize the terminal while staying scrolled up; verify the same anchored content stays in view and you don’t jump to bottom unexpectedly.
- Create a gap case (multiple non-continuation cells) and scroll so a blank spacer row is at/near the top; verify scrolling doesn’t get stuck on spacers and still anchors to nearby real lines.
- Start a selection while the assistant is streaming; verify the view stops auto-following, the selection stays on the intended content, and subsequent scrolling still behaves normally.
- Exit the TUI and confirm scrollback rendering still styles user rows as blocks (background padding) and non-user rows as expected.

joshka-oai committed:
fix: prep transcript scroll helpers

Move TranscriptScroll into tui::scrolling for isolated tests, shift scroll resolution/anchor logic into methods on TranscriptScroll, and keep explicit ToBottom assignments for intentional snaps while defaults handle initialization.

joshka-oai committed:
test: cover transcript scroll anchoring

Add rustdoc for scroll resolution and anchor helpers.
Add unit tests to lock in resolve_top/scrolled_by behavior.

joshka-oai committed:
refactor(tui2): make transcript line metadata explicit

Replace Option<(usize, usize)> transcript line metadata with TranscriptLineMeta::{CellLine, Spacer}.

This makes spacer rows explicit, avoids tuple semantics leaking through call sites, and keeps scroll anchoring documentation aligned with the viewport/history design notes.

Tests: cargo test -p codex-tui2 tui::scrolling

bolinfest reviewed (approved):
Please ensure CI is green before submitting.

joshka-oai committed:
fix(tui2): satisfy clippy redundant closure

Use a method reference (TranscriptLineMeta::cell_index) to satisfy clippy::redundant-closure-for-method-calls.

Clippy: cargo clippy -p codex-tui2 --all-features --tests --profile dev -- -D warnings
